neo/Network/P2P/ProtocolHandler.cs

```diff
@@ -21,9 +22,32 @@ public class SetVersion { public VersionPayload Version; }
         public class SetVerack { }
         public class SetFilter { public BloomFilter Filter; }
 
+        private class LimitedSet<T>
```

Comment:
Please move this data structure to its own file. It may be usedful other places as well.

Reply:
Also maybe rename to `LimitedFIFOSet`

Reply:
👍

Reply:
without inheritance?

Reply:
@shargon I think using composition is fine here rather than inheritance; however, it would probably be the best practice to have interface(s) it implements for its operations.